scripts/install_ebpf.psm1

```diff
@@ -219,48 +146,171 @@ function Install-eBPFComponents
+                "$pwd\drivers\{0}" -f $_.Value.Name
+            } else {
+                throw ("Driver file not found for $($_.Key).")
+            }
+            Write-Log("Installing $($_.Key)...") -ForegroundColor Green
+            sc.exe create $_.Key type=kernel start=demand binpath=$driverPath 2>&1 | Write-Log
+            if ($LASTEXITCODE -ne 0) {
+                throw ("Failed to create $($_.Key) driver.")
+            } else {
+                Write-Log("$($_.Key) driver created.") -ForegroundColor Green
+                # Start the service
+                Write-Log("Starting $($_.Key) service...") -ForegroundColor Green
+                sc.exe start $_.Key 2>&1 | Write-Log
+                if ($LASTEXITCODE -ne 0) {
+                    throw ("Failed to start $($_.Key) service.")
+                } else {
+                    Write-Log("$($_.Key) service started.") -ForegroundColor Green
+                }
+            }
+        }
+    }
+
+    # Export program info for the sample driver.
+    Write-Log("Running 'export_program_info_sample.exe'...")
+    if (Test-Path -Path "export_program_info_sample.exe") {
+        .\export_program_info_sample.exe 2>&1 | Write-Log
+        if ($LASTEXITCODE -ne 0) {
+            throw ("Failed to run 'export_program_info_sample.exe'.");
+        } else {
+            Write-Log "'export_program_info_sample.exe' succeeded." -ForegroundColor Green
+        }
     }
-    Copy-Item *.dll -Destination "$Env:systemroot\system32" -Force -ErrorAction Stop 2>&1 | Write-Log
-    Copy-Item *.exe -Destination "$Env:systemroot\system32" -Force -ErrorAction Stop 2>&1 | Write-Log
 
-    # Register all components.
-    Register-eBPFComponents
+    # Print the status of the eBPF drivers and services after installation.
+    Print-eBPFComponentsStatus "Verifing the status of eBPF drivers and services after the installation..." | Out-Null
 
+    # Optionally enable KMDF verifier and tag tracking.
     if ($KMDFVerifier) {
-        # Enable KMDF verifier and tag tracking.
         Enable-KMDFVerifier
     }
 
-    # Start all components.
-    Start-eBPFComponents -KmTracing $KmTracing -KmTraceType $KmTraceType
+    # Start KM tracing.
+    Start-WPRTrace -KmTracing $KmTracing -KmTraceType $KmTraceType
 }
 
-function Stop-eBPFComponents
+function Uninstall-eBPFComponents
 {
-    # Stop user mode service.
-    Stop-Service "eBPFSvc" -ErrorAction Ignore 2>&1 | Write-Log
-
-    # Stop the drivers.
+    # Firstly, uninstall the extra drivers that are not installed by the MSI package.
     $EbpfDrivers.GetEnumerator() | ForEach-Object {
-        Stop-Service $_.Name -ErrorAction Ignore 2>&1 | Write-Log
+        if (-not $_.Value.InstalledByMsi) {
+            Write-Log("Stopping $($_.Key) service...") -ForegroundColor Green
+            sc.exe stop $_.Key 2>&1 | Write-Log
+            if ($LASTEXITCODE -ne 0) {
+                Write-Log("Failed to stop $($_.Key) service.") -ForegroundColor Red
```

Comment:
agreed, now throwing exceptions.

Reply:
@dv-msft I have to push back on this, or rather remove the call to `Stop-eBPFComponents` happening in `execute_ebpf_cicd_tests.ps1`, which will otherwise cause the uninstall itself to fail.

Reply:
As discussed offline, we will ensure that we can stop all drivers and not being able to stop even one will be flagged as a fatal error.  The script(s) will be suitably modified to ensure this.

Reply:
I updated the previous `Stop-eBPFComponents` to throw exceptions, and the uninstall will rely (or fail) on that happening.